### PR TITLE
Added sync for the precursor key terminal

### PR DESCRIPTION
**Pull request by Coding-Hen**

### CODE REVIEW

File: NitroxModel/DataStructures/GameLogic/Entities/Metadata/PrecursorKeyTerminalMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+namespace NitroxModel.DataStructures.GameLogic.Entities.Metadata
+{
+    [Serializable]
+    [ProtoContract]
+    public class PrecursorKeyTerminalMetadata : EntityMetadata
+    {
+        [ProtoMember(1)]
+        public bool Slotted { get; }
+
+        public PrecursorKeyTerminalMetadata()
```
Comment (Jannify) [inline NitroxModel/DataStructures/GameLogic/Entities/Metadata/PrecursorKeyTerminalMetadata.cs:13]: This does not fit the implementation of other Metadata classes which can result in json saving problems. Pls look at other (up to date) Metadata classes.

File: NitroxClient/GameLogic/Spawning/Metadata/PrecursorKeyTerminalMetadataProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public class PrecursorKeyTerminalMetadataProcessor : GenericEntityMetadataProcessor<PrecursorKeyTerminalMetadata>
+    {
+        public override void ProcessMetadata(GameObject gameObject, PrecursorKeyTerminalMetadata metadata)
+        {
+            Log.Debug($"Received precursor key terminal metadata change for {gameObject.name} with data of {metadata}");
+
+            PrecursorKeyTerminal precursorKeyTerminal = gameObject.GetComponent<PrecursorKeyTerminal>();
+            if (precursorKeyTerminal)
+            {
+                Utils.PlayFMODAsset(precursorKeyTerminal.useSound, precursorKeyTerminal.transform, 20f);
```
Comment (Jannify) [inline NitroxClient/GameLogic/Spawning/Metadata/PrecursorKeyTerminalMetadataProcessor.cs:16]: As far as I remember this is also called when we players are joining. So this sound can falsely be played on connection (and if it's 2D not 3D everywhere in the world), am I right?

Comment (Coding-Hen) [reply NitroxClient/GameLogic/Spawning/Metadata/PrecursorKeyTerminalMetadataProcessor.cs:16]: Yes I beleive you are right on that one, it was something that crossed my mind anyway. Will remove till audio issues are fixed.

File: NitroxModel/DataStructures/GameLogic/Entities/Metadata/PrecursorKeyTerminalMetadata.cs (hunk 1 above)
Comment (Coding-Hen) [reply NitroxModel/DataStructures/GameLogic/Entities/Metadata/PrecursorKeyTerminalMetadata.cs:13]: By this are you refering to the protected decleration found on other metadata classes?

Comment (Jannify) [reply NitroxModel/DataStructures/GameLogic/Entities/Metadata/PrecursorKeyTerminalMetadata.cs]: Yes 😉

File: NitroxClient/GameLogic/Spawning/Metadata/PrecursorKeyTerminalMetadataProcessor.cs (hunk 2 above)
Comment (Jannify) [reply NitroxClient/GameLogic/Spawning/Metadata/PrecursorKeyTerminalMetadataProcessor.cs]: Ok. I am planning to work on a global audio syncing system once I get back home from vacation.

**Pull request merged**